docs/migrating.md

```diff
@@ -0,0 +1,116 @@
+# Migrating from Arm GNU Toolchain
+
+## Overview
+
+[Arm GNU Toolchain](https://developer.arm.com/Tools%20and%20Software/GNU%20Toolchain)
+is the GNU Toolchain for the Arm Architecture released by Arm and traditionally
+used for embedded development.
+
+Generally LLVM toolchain tries to be a drop in replacement for the GNU toolchain,
+however there may be some missing features or small incompatibilities.
+
+Some known differences and migration strategies are summarized below.
+
+## Benefits
+
+Using multiple toolchains to build a project benefits from different checks
+and warnings present in different compilers to catch more issues,
+particularly C and C++ standards compliance, during build time.
+
+LLVM Embedded Toolchain provides support for multiple sanitizers
+and memory safety features to also catch typical issues at runtime during testing,
+see the [LLVM documentation](https://clang.llvm.org/docs/index.html)
+and [samples](https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/tree/main/samples/src).
+
+LLVM Embedded Toolchain provides superior performance when targeting the
+Armv8-M or later architecture, including the
+[Arm Helium Technology](https://www.arm.com/technologies/helium)
+(M-Profile Vector Extension, MVE).
+
+## Key toolchain components
+
+|Component|GNU toolchain (`arm-none-eabi-`)|LLVM toolchain|
+|---------|-------------|--------------|
+|C, C++ compiler​|`gcc`, `g++`|`clang`, `clang++`​|​
+|Assembler​|`as`​|`clang` integrated assembler​|
+|Linker​|`ld`​|`lld`​|
+|Binutils​|`objdump`, `readelf`, ...|`llvm-objdump`, `llvm-readelf`, ...|
+|Compiler runtime library​|`libgcc​`|`compiler-rt`​|
+|Unwinder​|`libgcc`​|`libunwind`​|
+|C standard library​|`newlib`, `newlib-nano`|`picolibc`|​
+|C++ ABI library​|`libsupc++.a`|`libc++abi`​|​
+|C++ standard library​|`libstdc++​`|`libc++`​|
+
+## Toolchain identification
+
+Toolchain version macros:
+
+|Version number|GNU macro|LLVM macro|
+|-------|---------|----------|
+|Major|`__GNUC__`|`__clang_major__`|
+|Minor|`__GNUC_MINOR__`|`__clang_minor__`|
+|Patch level|`__GNUC_PATCHLEVEL__`|`__clang_patchlevel__`|
+
```

Comment:
Do we need some more explanation of the table?

Clang will define all of these, using 4, 2 and 1 (a very old GCC!) for the GNU defines. This is OK for code that checks for the existence, but not good for code that tries to detect a recent GCC to infer whether certain builtins exist.

Reply:
Very good point - this is an important detail, added.